### ISSUE TITLE
[java] UseArraysAsList should not warn when elements are skipped in array

### ISSUE THREAD

**Issue by tsmock** [pmd/pmd#5071]
**Affects PMD Version:** 7.2.0

**Rule:** UseArraysAsList

Please provide the rule name and a link to the rule documentation:
https://docs.pmd-code.org/latest/pmd_rules_java_performance.html#usearraysaslist

**Description:** UseArraysAsList should not warn when elements are skipped in array

**Code Sample demonstrating the issue:**

```java
import java.util.*;
public class Test {
  public static void main(String... args) {
    String[] entrySet = {"key1", "value1", "key2", "value2", "key3", "value3"};
    for (String key : keySet(entrySet)) {
      System.out.println(key);
    }
  }

  private static Collection<String> keySet(String... entrySet) {
    final List<String> result = new ArrayList<>(entrySet.length / 2);
    for (int i = 0; i < entrySet.length; i += 2) {
      result.add(entrySet[i]); // false positive here
    }
    return result;
  }
}
```

**Expected outcome:** No warning

PMD reports a violation at line 13, but that's wrong. That's a false positive.
Full message:
```
[INFO] PMD version: 7.2.0
[INFO] PMD Failure: Test:13 Rule:UseArraysAsList Priority:3 Use asList instead of tight loops.
```

**Running PMD through:** *Maven*

Possibly related to:
* #4014
* #4577
